CampusTour/TableCells/ItemOfInterestTableViewCell.swift

```diff
@@ -0,0 +1,237 @@
+    var locationLabel: UILabel?
+    var tagsStackView: UIStackView?
+    var wantedImageUrl: URL?
+    
+    func titleHeaderView(model: ModelInfo) -> UIView {
+        let header = UIStackView()
+        header.axis = .horizontal
+        
+        let titleLabel = UILabel.label(text: model.title,
+                                       color: Colors.primary,
+                                       font: UIFont.systemFont(ofSize: 16, weight: .semibold))
+        header.addArrangedSubview(titleLabel)
+        
+        return header
+    }
+    
+    func locationLineView(spec: LocationLineViewSpec) -> (UIView, UILabel) {
+        let hStack = UIStackView()
+        hStack.spacing = 8
+        hStack.axis = .horizontal
+        let label = UILabel.label(
+            text: spec.locationName,
+            color: Colors.secondary,
+            font: UIFont.systemFont(ofSize: 14))
+        hStack.addArrangedSubview(label)
+        return (hStack, label)
+    }
+    
+    func createLeftStackView(layout: Layout) -> UIStackView {
+        let leftStackView = UIStackView()
+        leftStackView.axis = .vertical
+        leftStackView.alignment = .leading
+        leftStackView.spacing = 8
+        
+        //add date label
+        if layout == .event {
+            dateLabel = UILabel.label(
+                text: "",
+                color: Colors.brand,
+                font: UIFont.systemFont(ofSize: 12, weight: .medium))
+            leftStackView.addArrangedSubview(dateLabel!)
+        }
+        
+        //add the title header
+        titleLabel = UILabel.label(
+            text: "",
+            color: Colors.primary,
+            font: UIFont.boldSystemFont(ofSize: 16))
+        leftStackView.addArrangedSubview(titleLabel!)
+        
+        //add location line
+        if layout == .event {
+            locationLabel = UILabel.label(text: "",
+                                      color: Colors.primary,
+                                      font: UIFont.systemFont(ofSize: 14))
+            leftStackView.addArrangedSubview(locationLabel!)
+        }
+        
+        tagsStackView = tagsHStackView(tags: [])
+        leftStackView.addArrangedSubview(tagsStackView!)
+        
+        return leftStackView
+    }
+    
+    func createRightView(layout: Layout) -> UIView {
+        let rightView = UIStackView()
+        rightView.axis = .vertical
+        
+        itemImageView = UIImageView()
+        imageView?.contentMode = .scaleAspectFill
+        itemImageView?.layer.cornerRadius = 8
+        itemImageView?.clipsToBounds = true
+        itemImageView?.snp.makeConstraints { make in
+            make.width.equalTo(84)
+            make.height.equalTo(48)
+        }
+        rightView.addArrangedSubview(itemImageView!)
+        
+        rightView.addArrangedSubview(UIView())
+        
+        return rightView
+    }
+    
+    func setUpViewsIfNecessary(layout: Layout) {
+        if self.currentLayout == layout { return }
```

Comment:
Do we need this function? this is for optimization when we are reusing the cells with the same layout?

Reply:
yeah, it's if the layout is the same. Since the same class is used for 2 different layouts, I made this function